src/OmniSharp.MSBuild/ProjectLoader.cs

```diff
@@ -127,10 +131,7 @@ private static void SetTargetFrameworkIfNeeded(MSB.Evaluation.Project evaluatedP
                 // do better and potentially allow OmniSharp hosts to select a target framework.
                 targetFramework = targetFrameworks[0];
                 evaluatedProject.SetProperty(PropertyNames.TargetFramework, targetFramework);
-            }
-            else if (!string.IsNullOrWhiteSpace(targetFramework) && targetFrameworks.Length == 0)
```

Comment:
This check is not needed here. We used to pass `targetFrameworks` as an out parameter but that was removed long ago. So, this is essentially unnecessary code.